clients/python-wrapper/lakefs/branch.py

```diff
@@ -236,12 +238,18 @@ def revert(self, reference_id: str, parent_number: Optional[int] = None) -> None
         elif parent_number <= 0:
             raise ValueError("parent_number must be a positive integer")
 
+        # Handle reference_id as a deprecated alias to reference.
+        reference = reference or reference_id
+        ref = reference if isinstance(reference, str) else reference.id
+
         with api_exception_handler():
-            return self._client.sdk_client.branches_api.revert_branch(
+            self._client.sdk_client.branches_api.revert_branch(
                 self._repo_id,
                 self._id,
-                lakefs_sdk.RevertCreation(ref=reference_id, parent_number=parent_number)
+                lakefs_sdk.RevertCreation(ref=ref, parent_number=parent_number)
             )
+            commit = self._client.sdk_client.commits_api.get_commit(self._repo_id, self._id)
```

Comment:
I think this is still a race: multiple processes may be working on the same branch.  The field `self._id` is the name of a branch, for instance `main`.  Suppose that I am running this code to revert, and concurrently Nir merges into the branch.  My code runs revert and moves `main`, now Nir merges in and moves `main` again, now my code runs get_commit.  I get Nir's merge, which is **not** the result of the merge.  That's why the revert API returns a new digest: it's the only way for the caller to know what it created!


I think that you will need something like this:
```suggestion
            new_digest = self._client.sdk_client.branches_api.revert_branch(
                self._repo_id,
                self._id,
                lakefs_sdk.RevertCreation(ref=reference_id, parent_number=parent_number)
                lakefs_sdk.RevertCreation(ref=ref, parent_number=parent_number)
            )
            commit = self._client.sdk_client.commits_api.get_commit(self._repo_id, new_digest)
```

Reply:
As you rightly explained, there is a bug in our SDK: revertBranch does not return the generated commit.  So there is no way to implement this correctly on the client.  Given this, let's go with what we can do.  I shall open a bug to work around this, and future versions of client and server may be able to do this.

Please add words to the docstring such as
> ```
> Due to a limitation of lakeFS IO, the returned commit might be incorrect when
> called on the server concurrently with other operations that affect the commit
> log, such as commits, merges, cherry-picks, and reverts.
> ```

Other than that, we can pull.

THANKS!